remove metalium api violation - move fabric packet header out

SeanNijjar opened the pull request:
this change removes the Fabric packet header related Metalium API leaks. The packet header... header was exposed through Metalium host APIs but really shouldn't have been. 

Closes #20621 


### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/17195356333
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI: https://github.com/tenstorrent/tt-metal/actions/runs/17196178569
- [x] T3K: hhttps://github.com/tenstorrent/tt-metal/actions/runs/17196186595
  - [x] unit
  - [x] demo: compare with https://github.com/tenstorrent/tt-metal/actions/runs/17195497213 and https://github.com/tenstorrent/tt-metal/actions/runs/17216996969
  - [x] frequent: same failures (https://github.com/tenstorrent/tt-metal/actions/runs/17196183645/job/48778759994)
  - [x] nightly 
- [x] BH nightly: https://github.com/tenstorrent/tt-metal/actions/runs/17197364104

afuller-TT commented on setup.py:
```diff
@@ -294,7 +294,7 @@ def run(self) -> None:
             "api/tt-metalium/dev_msgs.h",
             "api/tt-metalium/fabric_edm_types.hpp",
-            "api/tt-metalium/fabric_edm_packet_header.hpp",
+            "fabric/fabric_edm_packet_header.hpp",
```
If it's moved OUT of the public API, why does it need to get packaged up?

Or is it only out of the public API host-side and it's still in the API device-side?

afuller-TT replied:
Update: Looks like it's a JIT-header now.

afuller-TT commented on tt_metal/impl/profiler/profiler.cpp:
```diff
@@ -191,11 +191,6 @@ void sortDeviceEvents(std::vector<std::reference_wrapper<const tracy::TTDeviceEv
         });
 
-    for (uint32_t i = 0; i < (num_threads / 2) - 1; ++i) {
-        threads[i].join();
```
Is this related?  Or snuck in?

SeanNijjar replied:
hmm... not sure how that got there. I'm going to pull this into draft and review everything again

afuller-TT commented:
Looks like kernels like https://github.com/tenstorrent/tt-metal/blob/main/tt_metal/impl/dispatch/kernels/cq_relay.hpp#L10 will need to learn the path to the new home.  Previously the JIT Build considered api/tt-metalium as an include search path.  No such "known location" exists for component-specific pieces like Fabric.

SeanNijjar commented on tests/ttnn/unit_tests/gtests/ccl/test_fabric_edm_common.hpp:
```diff
@@ -467,479 +466,6 @@ struct mcast_send {
 
 using mode_variant_t = std::variant<mcast_send, unicast_send>;
 
-
```
deleting dead test code

SeanNijjar commented on tests/ttnn/unit_tests/gtests/ccl/test_fabric_erisc_data_mover_loopback_with_workers.cpp:
```diff
@@ -69,358 +68,6 @@
 ////////////////////////////////////////////////////////////////////
 
-TEST(WorkerCclCommandProcessingKernelLocalMode, DISABLED_MultiInputReader_SinglePageTile) {
-    auto pass = RunMultiInputReaderTestPropagateFullTensorIn(
```
Deleting dead tests for unused feature (FYI @ntarafdar)

SeanNijjar commented on ttnn/cpp/ttnn/operations/ccl/common/uops/ccl_command.hpp:
```diff
@@ -576,12 +575,10 @@ enum class CclCommandCode : uint8_t {
 
 
 enum CclCommandDestType : uint8_t {
```
this is all dead code and should be deleted (but wasn't deleted here because it wasn't blocking the core changes here)

afuller-TT commented:
The smoke tests are failing because the header is not being packaged up for the JIT build.  The kernel is reaching out into the void and finding nothing.

https://github.com/tenstorrent/tt-metal/actions/runs/16898491191/job/47873589817?pr=26635#step:7:64

jvegaTT reviewed (approved):
Nice cleanup

SeanNijjar commented on tt_metal/fabric/fabric_edm_packet_header.hpp:
```diff
@@ -313,7 +313,8 @@ struct PacketHeaderBase {
                 edm_to_local_chip_noc);
```
probably that's the right final resting place but I believe there is still control plane code that needs access to this. This PR is mainly looking just at moving the file out of `api` but we should further clean things up on metal side to split the code into host interface and device impl portions.

blozano-tt commented on setup.py:
```diff
@@ -289,7 +289,7 @@ def run(self) -> None:
             "api/tt-metalium/dev_msgs.h",
             "api/tt-metalium/fabric_edm_types.hpp",
-            "api/tt-metalium/fabric_edm_packet_header.hpp",
+            "tt_metal/fabric/fabric_edm_packet_header.hpp",
```
Not sure the `tt_metal` prefix is needed here.

See how there are other files from the `fabric` directory bundled a few lines below.

SeanNijjar replied:
how can I go about testing a wheel build offline/in workspace so I don't need to have a roundtrip to CI to resolve the kernel build issues I'm seeing there (but not locally)

SeanNijjar replied:
never mind this is resolved now!

SeanNijjar committed:
update include path

SeanNijjar committed:
update path

SeanNijjar committed:
fix after rebase

SeanNijjar committed:
oops

SeanNijjar committed:
restore include path

SeanNijjar committed:
fixes after rebase

SeanNijjar committed:
?

SeanNijjar committed:
update include paths

SeanNijjar committed:
fix device side include paths

SeanNijjar committed:
WIP: move packet header out of API :D

Update packet header inclusion paths or remove inclusion (WIP)

move erisc datamover builder out of metal API, update include paths

compiling

cleanup

missed one

yikes

SeanNijjar committed:
include packet header in jit api sources

SeanNijjar committed:
fix include path

SeanNijjar committed:
guys... stop...
